File: maze-walker/typescript/src/MazeExceptions.ts
export class NoStartCellException extends Error {
  constructor() {
    super('Maze must have exactly one start cell.');
    this.name = 'NoStartCellException';
  }
}

export class NoEndCellException extends Error {
  constructor() {
    super('Maze must have exactly one end cell.');
    this.name = 'NoEndCellException';
  }
}

export class MultipleStartCellsException extends Error {
  constructor() {
    super('Maze must have exactly one start cell.');
    this.name = 'MultipleStartCellsException';
  }
}

export class MultipleEndCellsException extends Error {
  constructor() {
    super('Maze must have exactly one end cell.');
    this.name = 'MultipleEndCellsException';
  }
}

File: maze-walker/python/src/maze_walker/maze_exceptions.py
class NoStartCellException(Exception):
    def __init__(self) -> None:
        super().__init__("Maze must have exactly one start cell.")


class NoEndCellException(Exception):
    def __init__(self) -> None:
        super().__init__("Maze must have exactly one end cell.")


class MultipleStartCellsException(Exception):
    def __init__(self) -> None:
        super().__init__("Maze must have exactly one start cell.")


class MultipleEndCellsException(Exception):
    def __init__(self) -> None:
        super().__init__("Maze must have exactly one end cell.")
